### ISSUE TITLE
GL Conflict Between this and OpenModsLib

### ISSUE THREAD

**Issue by Parker8283** [CoFH/Feedback#308]
With both CoFH Core 3.0.0RC7 and OpenModsLib 0.6 installed, I receive the following error:
<pre>[22:41:54] [Client thread/ERROR]: ########## GL ERROR ##########
[22:41:54] [Client thread/ERROR]: @ Pre startup
[22:41:54] [Client thread/ERROR]: 1286: Invalid framebuffer operation</pre>
This causes nothing to draw on screen ever. This happens once when the Mojang logo is trying to be drawn, and then it gets spammed in the console when trying to draw the main menu.

Also reported here: https://github.com/OpenMods/OpenModsLib/issues/43

**Comment by boq** [comment]
Some context on from my side:

Here's my code: https://github.com/OpenMods/OpenModsLib/blob/master/src/main/java/openmods/stencil/FramebufferHooks.java. I'm replacing original invocation to `OpenGlHelper.func_153186_a` with `FramebufferHooks.createRenderbufferStorage`, so I can check GL error and just degrade to legacy functionality.

It looks like [your code](https://github.com/CoFH/CoFHCore/blob/master/src/main/java/cofh/core/ProxyClient.java#L114) is doing what I tried to do in MinecraftForge/MinecraftForge#1465 (i.e. enable stencil for FBO, while keeping legacy stencil buffer disabled).

Though I'm not sure if it even works, since `GL11.glGetInteger(GL11.GL_STENCIL_BITS);` is actual value (not capability one) - and (at least on my computer) is always 0 when `forge.forceDisplayStencil` is not set.

Actually, I can't even reproduce original issue, so it may be yet another broken driver.
**tl;dr;** I have no idea what's happening

BTW, PR was closed, since there is small number configurations (about 1-2% of GPUs according to various sites) where even that was broken. Problem with Forge code is that [here](https://github.com/MinecraftForge/MinecraftForge/blob/1.7.10/patches/minecraft/net/minecraft/client/shader/Framebuffer.java.patch#L14) - `EXTPackedDepthStencil` may be used even when it's not available (`EXTFramebufferObject` is not enough). I see you try to avoid that pitfall in your code with `glCheckFramebufferStatusEXT` call .

**Comment by boq** [comment]
Ok, I've hacked very dirty mod to log all GL calls and errors (result in parallel thread on OpenModsLib) and found out that `EXTFramebufferObject.glCheckFramebufferStatusEXT(target)` is used incorrectly. According to specification, "target must be `GL_DRAW_FRAMEBUFFER`, `GL_READ_FRAMEBUFFER` or `GL_FRAMEBUFFER`", and not actual object id. So this call always fails with `GL_INVALID_ENUM`.

Still, I have no idea how it interferes with my code. Since this only works when Minecraft is using EXT stencil objects, it's very hard to test without specific software.

**Comment by skyboy** [comment]
I've fixed those erroneous calls, so maybe that'll resolve the issue. Also: [this](http://support.elmark.com.pl/rgd/drivery/U12C/Graphic/Win_7_64/releasenotes_gfx.htm) is a list of the affected cards, and the specific driver causing the problem. It appears related to bug 3356195, but in OGL support instead of DX.

**Comment by Parker8283** [comment]
I have ATI Raedon

**Comment by skyboy** [comment]
Oh great, there are more. At least you can check to see if the changes I've made fix it or not, so I can make more changes or not.

**Comment by Parker8283** [comment]
I will tonight when I get time. This may also be due to the fact that my drivers are wildly out of date and, because I'm running Mac, I can't update them. Potentially. Not sure. Also, I'm running OpenGL 2.1.

**Comment by Parker8283** [comment]
Could you push the alpha with the fix to CurseForge please?

**Comment by Parker8283** [comment]
Self-compiled the code with the fix, and it runs just fine alongside OpenModsLib. While there may be some things left on their side, the original issue is now fixed on your side, so I'm going to consider this closed. Thanks for the help!
